### DISCUSSION TITLE
Question About Model Parameter Updates（self.training=false & requires_grad=true）

### DISCUSSION THREAD

**Discussion by D222097** [Lightning-AI/pytorch-lightning#21329] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
What will happen when self.vlm.training is set to False and self.vlm.parameters() has requires_grad set to True? Will the model parameters be updated if there is no dropout or batch normalization in the model?

Besides, I'm opening this issue due to an unexpected change in my model. The model includes self.dit, self.vlm, and other components. After completing self.train(), self.vlm.training is set to True, but when I enter training_step(), it switches to False. I'm not sure what is causing this behavior. Any suggestions regarding the potential reasons for this issue would be greatly appreciated.
```
def on_validation_end(self):
        self.train()
        # self.vlm.training is True
        # self.dit.training is True
```
```
def training_step(self, batch: Dict[str, Dict], batch_idx: int) -> torch.Tensor:
        # self.vlm.training is False
        # self.dit.training is True
        opt = self.optimizers()
        ...
```
